lib/improver/threshold.py

```diff
@@ -110,32 +119,41 @@ def process(self, input_cube):
             ValueError: if a np.nan value is detected within the input cube.
 
         """
-        cube = input_cube.copy()
-        if np.isnan(cube.data).any():
+        thresholded_cubes = iris.cube.CubeList()
+        if np.isnan(input_cube.data).any():
             raise ValueError("Error: NaN detected in input cube data")
-        if self.fuzzy_factor is None:
-            truth_value = cube.data > self.threshold
-        else:
-            lower_threshold = self.threshold * self.fuzzy_factor
-            truth_value = (
-                (cube.data - lower_threshold) /
-                ((self.threshold * (2. - self.fuzzy_factor)) - lower_threshold)
-            )
-        truth_value = np.clip(truth_value, 0., 1.).astype(np.float64)
-        if self.below_thresh_ok:
-            truth_value = 1. - truth_value
-        cube.data = truth_value
-
-        # TODO: Correct when formal cf-standards exists
-        # Force the metadata to temporary conventions
-        if self.below_thresh_ok:
-            cube.attributes.update({'relative_to_threshold': 'below'})
-        else:
-            cube.attributes.update({'relative_to_threshold': 'above'})
-        cube.rename("probability_of_{}".format(cube.name()))
-        coord = iris.coords.DimCoord(self.threshold,
-                                     long_name="threshold",
-                                     units=cube.units)
-        cube.add_aux_coord(coord)
-        cube.units = Unit(1)
+
+        for threshold in self.thresholds:
+            cube = input_cube.copy()
+            if self.fuzzy_factor is None:
+                truth_value = cube.data > threshold
+            else:
+                lower_threshold = threshold * self.fuzzy_factor
+                truth_value = (
+                    (cube.data - lower_threshold) /
+                    ((threshold * (2. - self.fuzzy_factor)) - lower_threshold)
+                )
+            truth_value = np.clip(truth_value, 0., 1.).astype(np.float64)
+            if self.below_thresh_ok:
+                truth_value = 1. - truth_value
+            cube.data = truth_value
+
+            # TODO: Correct when formal cf-standards exists
```

Comment:
These attributes will always apply to every threshold slice in the cube, so can be moved outside the loop to after the concatenate statement.

Reply:
Moved.